libcontainer/mount_linux.go

```diff
@@ -47,19 +48,31 @@ func (e *mountError) Unwrap() error {
 	return e.err
 }
 
-// mount is a simple unix.Mount wrapper. If procfd is not empty, it is used
-// instead of target (and the target is only used to add context to an error).
-func mount(source, target, procfd, fstype string, flags uintptr, data string) error {
+// mount is a simple unix.Mount wrapper, returning an error with more context
+// in case it failed.
+func mount(source, target, fstype string, flags uintptr, data string) error {
+	return mountWithFD(source, "", target, "", fstype, flags, data)
+}
+
+// mountWithFD is a unix.Mount wrapper which uses srcFD instead of source, and
+// dstFD instead of target, unless those are empty.  In such cases, source and
+// target are only used to add context in case the mount failed.
+func mountWithFD(source, srcFD, target, dstFD, fstype string, flags uintptr, data string) error {
```

Comment:
I wonder if we want a `mountWithDstFD()` in addition, as we use that a lot too. No strong opinion

Reply:
I thought about it and decided it's too much.

One other way to solve this would be to refactor this to use a structure instead of list of parameters, but again I ended up not doing it.